hepa-uv/firmware/hepauv_hardware.c

```diff
@@ -162,13 +196,18 @@ static void MX_TIM_Init(TIM_TypeDef* tim) {
 
     htim->State = HAL_TIM_STATE_RESET;
     htim->Instance = tim;
-    /*
-     * Setting counter clock frequency to 2 kHz
-     */
-    htim->Init.Prescaler =
-        calc_prescaler(SystemCoreClock, LED_TIMER_FREQ);
     htim->Init.CounterMode = TIM_COUNTERMODE_UP;
-    htim->Init.Period = LED_PWM_WIDTH - 1;
+
+    // Set prescaler and period for hepa_pwm timer (TIM3)
+    if (tim == TIM3) {
+        htim->Init.Prescaler = 64 - 1;
+        htim->Init.Period = 40 - 1;
```

Comment:
this is not a lot of control authority for this PWM, I think. You'll only be able to have 40 levels. Maybe it's fine but something to be aware of